webv2/primarykey/hotspot.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package web defines web APIs to be used with harbourbridge frontend.
+// Apart from schema conversion, this package involves API to update
+// converted schema.
+
+package primarykey
+
+import (
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/cloudspannerecosystem/harbourbridge/webv2/session"
+)
+
+// DetectHotspot add hotspot suggetion in schema conversion process for schema.
+func DetectHotspot() {
+
+	sessionState := session.GetSessionState()
+
+	for _, spannerTable := range sessionState.Conv.SpSchema {
+
+		isHotSpot(spannerTable.Pks, spannerTable)
+	}
+
+}
+
+//Helpers method for hotspot detection.
+func isHotSpot(insert []ddl.IndexKey, spannerTable ddl.CreateTable) {
+
+	hotspotTimestamp(insert, spannerTable)
+	hotspotAutoincrement(insert, spannerTable)
+}
+
+// hotspotTimestamp check Timestamp hotspot.
+// if prseent add Hotspot_Timestamp as issue in Issues.
```

Comment:
// hotspotTimestamp checks Timestamp hotspot.
// If present adds Hotspot_Timestamp as an issue in Issues.